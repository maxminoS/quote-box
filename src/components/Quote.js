import React from "react";

export default props => (
  <>
    {props.quote
     ? <>
         <h3>{props.quote.quote}</h3>
         <p>- {props.quote.author}</p>
       </>
     : ""}
  </>
);
