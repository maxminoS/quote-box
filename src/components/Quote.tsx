import React from "react";
import { Typography } from "@material-ui/core";
import { FormatQuote } from "@material-ui/icons";

type Props = {
  quote: {
    quote: string,
    author: string,
  }
}

export const Quote = (props: Props) => (
  <>
    {props.quote
     ? <>
         <FormatQuote fontSize="large" />
         <Typography variant="h6">
           {props.quote.quote}
         </Typography>
         <Typography variant="subtitle1"
                     align="right"
         >
           - {props.quote.author}
         </Typography>
       </>
     : ""}
  </>
);
