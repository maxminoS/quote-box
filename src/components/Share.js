import React from "react";
import { IconButton } from '@material-ui/core';
import { Twitter } from '@material-ui/icons';

export default props => (
  <>
    {props.quote
     ? <IconButton onClick={() => window.open("https://twitter.com/intent/tweet?text=" + encodeURIComponent("\"" + props.quote.quote + "\"\n- " + props.quote.author))
                           }
       >
         <Twitter color="primary" fontSize="large" />
       </IconButton>
     : ""}
  </>
);
