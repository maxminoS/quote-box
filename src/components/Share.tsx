import React from "react";
import { IconButton } from '@material-ui/core';
import { Twitter } from '@material-ui/icons';

type Props = {
  quote: {
    quote: string,
    author: string,
  }
}

export default (props: Props) => (
  <>
    {props.quote
     ? <IconButton onClick={() => window.open("https://twitter.com/intent/tweet?text=" + encodeURIComponent("\"" + props.quote.quote + "\"\n- " + props.quote.author))
                           }
       >
         <Twitter color="primary" fontSize="large" />
       </IconButton>
     : ""}
  </>
);
