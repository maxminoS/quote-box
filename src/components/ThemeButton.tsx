import React from "react";
import { Grid, Tooltip, IconButton } from '@material-ui/core';
import { Brightness3, WbSunny } from "@material-ui/icons";

type Props = {
  darkMode: boolean,
  setDarkMode: React.Dispatch<React.SetStateAction<boolean>>
}

export const ThemeButton = (props: Props) => (
  <Grid item
        style={{
          position: "absolute",
          bottom: 8,
          right: 8
        }}
  >
    <Tooltip title={ props.darkMode ? "Dark Mode" : "Light Mode"} placement="left" arrow>
      <IconButton onClick={() => props.setDarkMode(!props.darkMode)}>
        { props.darkMode ? <Brightness3 /> : <WbSunny /> }
      </IconButton>
    </Tooltip>
  </Grid>
);
