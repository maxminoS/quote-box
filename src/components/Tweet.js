import React from "react";

export default props => (
  <>
    {props.quote
     ? <a href={"https://twitter.com/intent/tweet?text="
                + escape("\"" + props.quote.quote + "\"\n- " + props.quote.author)
               }
         target="_blank"
       >
         Tweet
       </a>
     : ""}
  </>
);
