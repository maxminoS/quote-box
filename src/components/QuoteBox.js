import React, { useState, useEffect } from "react";
import { Button } from "@material-ui/core";
import { Card, CardContent, CardActions } from "@material-ui/core";

import Quote from "components/Quote";
import Share from "components/Share";
import QuoteApi from "api/QuoteApi";

export default () => {
  const [quotes, setQuotes] = useState([]);
  const [quoteNumber, setQuoteNumber] = useState(0);

  useEffect(async () => {
    QuoteApi.getQuote()
      .then(quotes => {
        setQuotes(quotes);
        setQuoteNumber(Math.floor(Math.random() * quotes.length));
      });
  }, []);

  const handleClick = () => {
    setQuoteNumber(Math.floor(Math.random() * quotes.length));
  };

  return (
    <Card raised="true">
      <CardContent>
        <Quote
          quote={quotes[quoteNumber]}
        />
      </CardContent>
      <CardActions>
        <Share
          quote={quotes[quoteNumber]}
        />
        <Button variant="contained" color="secondary" style={{marginLeft: 'auto'}} onClick={handleClick}>
          GET
        </Button>
      </CardActions>
    </Card>
  );
}
