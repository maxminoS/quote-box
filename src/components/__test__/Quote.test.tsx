import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Quote } from 'components/Quote';

it('renders Quote component correctly', () => {
  render(<Quote quote={{
    "quote": "a quote",
    "author": "by this"
  }} />);

  expect(screen.getByRole('heading', { name: /a quote/i })).toBeInTheDocument();
  expect(screen.getByRole('heading', { name: /\- by this/i })).toBeInTheDocument();
});

it('renders Quote component as expected', () => {
  const quote = renderer.create(<Quote quote={{
    "quote": "a quote",
    "author": "by this"
  }} />).toJSON();
  expect(quote).toMatchSnapshot();
});
