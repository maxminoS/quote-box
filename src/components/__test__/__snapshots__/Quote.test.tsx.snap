// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Quote component as expected 1`] = `
Array [
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
    />
  </svg>,
  <h6
    className="MuiTypography-root MuiTypography-h6"
  >
    a quote
  </h6>,
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignRight"
  >
    - 
    by this
  </h6>,
]
`;
